R/estimatr_lh_robust.R

```diff
@@ -0,0 +1,184 @@
+#' @param subset An optional bare (unquoted) expression specifying a subset
+#' of observations to be used.
+#' @param clusters An optional bare (unquoted) name of the variable that
+#' corresponds to the clusters in the data.
+#' @param fixed_effects An optional right-sided formula containing the fixed
+#' effects that will be projected car_lht of the data, such as \code{~ blockID}. Do not
+#' pass multiple-fixed effects with intersecting groups. Speed gains are greatest for
+#' variables with large numbers of groups and when using "HC1" or "stata" standard errors.
+#' See 'Details'.
+#' @param se_type The sort of standard error sought. If \code{clusters} is
+#' not specified the options are "HC0", "HC1" (or "stata", the equivalent),
+#'  "HC2" (default), "HC3", or
+#' "classical". If \code{clusters} is specified the options are "CR0", "CR2" (default), or "stata". Can also specify "none", which may speed up estimation of the coefficients.
+#' @param ci logical. Whether to compute and return p-values and confidence
+#' intervals, TRUE by default.
+#' @param alpha The significance level, 0.05 by default.
+#' @param return_vcov logical. Whether to return the variance-covariance
+#' matrix for later usage, TRUE by default.
+#' @param try_cholesky logical. Whether to try using a Cholesky
+#' decomposition to solve least squares instead of a QR decomposition,
+#' FALSE by default. Using a Cholesky decomposition may result in speed gains, but should only
+#' be used if users are sure their model is full-rank (i.e., there is no
+#' perfect multi-collinearity)
+#' @param linearHypothesis A character string or a matrix specifying combination.
+#' See \code{\link[car]{linearHypothesis}} for more details.
+#' @details
+#'
+#' This function is a wrapper for \code{\link{lm_robust}} and for
+#' \code{\link[car]{linearHypothesis}}. It first runs \code{lm_robust} and
+#' next passes \code{"lm_robust"} object as an argument to \code{linearHypothesis}.
+#'
+#' @return An object of class \code{"lh_robust"} containing the two following components:
+#'
+#' \item{lm_robust}{an object as returned by \code{lm_robust}}.
+#' \item{lh}{A data frame with most of its columns pulled from \code{linearHypothesis}' output.}
+#'
+#' The only analyis directly performed by \code{lh_robust} is a \code{t-test} for the null hypothesis of no effects of the linear combination of coefficients as specified by the user.
+#' All other output components are either extract from \code{linearHypothesis} or \code{lm_robust}.
+#'
+#' The original output returned by  \code{linearHypothesis} is added as an attribute.
+#'
+#' @examples
+#'
+#' library(fabricatr)
+#' dat <- fabricate(
+#'   N = 40,
+#'   y = rpois(N, lambda = 4),
+#'   x = rnorm(N),
+#'   z = rbinom(N, 1, prob = 0.4),
+#'   clusterID = sample(1:4, 40, replace = TRUE)
+#' )
+#'
+#' # Default variance estimator is HC2 robust standard errors
+#' lhro <- lh_robust(y ~ x + z, data = dat, linearHypothesis = "z + 2x = 0")
```

Comment:
The example and example comment shouldn't focus on the SE estimtor but rather the specification of the linearHypothesis, and it should show several different possible linearHypotheses.

Reply:
ok makes sense. Kept both